service/history/api/recordactivitytaskstarted/api.go

```diff
@@ -54,7 +56,7 @@ func Invoke(
 	namespace := namespaceEntry.Name()
 
 	response := &historyservice.RecordActivityTaskStartedResponse{}
-	var dropTask bool
+	var startedTransition bool
```

Comment:
uh oh, this was a mess to merge before since the logic got extracted. see what I did in the merge commit and if you like that or want to do it differently